index.html

```diff
@@ -159,141 +156,401 @@ <h2>
-          </p>
-          <p>
-            Once in the locked state the <a>user agent</a> must fire all
-            relevant user generated {{MouseEvent}} events (for example:
-            `mousemove`, `mousedown`, `mouseup`, `click`, and `wheel`)
-            [[ui-events]] to the <a>target</a> of pointer lock, and not fire
-            mouse events to other elements. Events that require the concept of
-            a mouse cursor must not be dispatched (for example: `mouseover`,
-            `mouseout`, `drag`, and `drop`).
-          </p>
-          <p>
-            In the locked state the system mouse cursor must be hidden.
-            Movement and button presses of the mouse must not cause the window
-            to lose focus.
-          </p>
-          <p>
-            Synthetic mouse events created by application script act the same
-            regardless of lock state.
+            When <dfn>requestPointerLock()</dfn> invoked, perform the following
+            steps:
           </p>
+          <ol class="algorithm">
+            <li>Let |promise:Promise| be [=a new promise=].
+            </li>
+            <li>When a window is in focus, if the [=this=]'s [=shadow-including
+            root=] is the [=navigable/active document=] of a [=Document/browsing
+            context=] (or has an [=tree/ancestor=] [=browsing context=]) which is not in
+            focus:
+              <ol>
+                <li>[=Queue an element task=] on the [=user interaction task
+                source=], given [=this=], to perform the following steps:
+                  <ol>
+                    <li>[=Fire an event=] named {{pointerlockerror}} at
+                    [=this=]'s [=Node/node document=].
+                    </li>
+                    <li>[=Reject=] |promise| with a
+                    {{"WrongDocumentError"}} {{DOMException}}.
+                    </li>
+                  </ol>
+                </li>
+                <li>Return <var>promise</var>.
+                </li>
+              </ol>
+            </li>
+            <li>If the user agent's [=pointer-lock target=] is an element whose
+            [=shadow-including root=] is not equal to [=this=]'s
+            [=shadow-including root=], then:
+              <ol>
+                <li>[=Queue an element task=] on the [=user interaction task
+                source=], given this, to perform the following steps:
+                  <ol>
+                    <li>[=Fire an event=] named {{pointerlockerror}} at
+                    [=this=]'s [=Node/node document=].
+                    </li>
+                    <li>[=Reject=] <var>promise</var> with a
+                    "{{InvalidStateError}}" {{DOMException}}.
+                    </li>
+                  </ol>
+                </li>
+                <li>Return <var>promise</var>.
+                </li>
+              </ol>
+            </li>
+            <li>If the user agent's [=pointer-lock target=]'s
```

Comment:
why don't just have flag to prevent you call the method twice? It seems complicated to check for object equality. 

What's the actual use case for calling the method twice? Why not unlock and lock again?

Reply:
@james-howard provides a good use case of repeated requestPointerLock requests with different options. 
If that also sounds legit to you, may I mark this comment resolved?

Reply:
Sorry, I think we might be talking past each other. 

I agree that the following use case is totally valid:
```JS
// Async switching...
await elem.requestPointerLock(options);
await elem.requestPointerLock(differentOptions);
```

I don't agree that this case (sync) is valid or useful: 

```JS
const p1 = elem.requestPointerLock(options);
const p2 = elem.requestPointerLock(differentOptions);
```

I think p2 (and any subsequent call) should reject unless `p1` has settled.

Reply:
Let's pick this up below... it's the same question...

Reply:
(Yes let's discuss this below on https://github.com/w3c/pointerlock/pull/49/files#r1373662771)